Apps/Src/Print_Queue.c

```diff
@@ -0,0 +1,69 @@
+/* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
+/** Print_Queue.c
+ * Queue that holds all characters to be printed that Task_Print needs to send to UART.
+ */
+
+#include "Print_Queue.h"
+#include "RTOS_BPS.h"
+
+#define FIFO_TYPE char
+#define FIFO_SIZE (128)
+#define FIFO_NAME Print_Fifo
+#include "fifo.h"
+
+static Print_Fifo_t printFifo;
+static OS_SEM printFifo_Sem;
+
+/**
+ * @brief Initializes the print queue
+ * @param none
+ * @return none
+ */
+void Print_Queue_Init() {
+    Print_Fifo_renew(&printFifo);
+    RTOS_BPS_SemCreate(&printFifo_Sem, "printSem", 0);
+}
+
+bool New_Line_Check();
+
+/**
+ * @brief Initializes the print queue
+ * @param buffer String of formatted text to be added to the buffer
+ * @param len Length of formatted string
+ * @return If the write was successful (there was room in the buffer)
+ */
+bool Print_Queue_Append(char *buffer, unsigned int len) {
+    int i = 0;
+    while(i < len){
+        Print_Fifo_put(&printFifo, buffer[i]);
```

Comment:
need to acquire a fifo mutex before appending. this should probably be done on a thread by thread basis to avoid splitting messages in the middle (which is the whole reason why we're doing this library -- to avoid this).